tests/test_predicates.py

```diff
@@ -126,3 +127,90 @@ def test_disjoint():
     a2 = spherely.point(50, 9)
     b2 = spherely.linestring([(50, 8), (60, 8)])
     assert spherely.disjoint(a2, b2)
+
+
+def test_touches():
+    a = spherely.polygon([(0.0, 0.0), (0.0, 1.0), (1.0, 1.0), (1.0, 0.0)])
+    b = np.array(
+        [
+            spherely.polygon([(1.0, 1.0), (1.0, 2.0), (2.0, 2.0), (2.0, 1.0)]),
+            spherely.polygon([(0.5, 0.5), (0.5, 1.5), (1.5, 1.5), (1.5, 0.5)]),
+        ]
+    )
+
+    actual = spherely.touches(a, b)
+    expected = np.array([True, False])
+    np.testing.assert_array_equal(actual, expected)
+
+    a_p = spherely.point(1.0, 1.0)
+    b_p = spherely.point(1.0, 1.0)
+    # Points do not have a boundary, so they cannot touch per definition
+    # This is consistent with PostGIS for example (cmp. https://postgis.net/docs/ST_Touches.html)
+    assert not spherely.touches(a_p, b_p)
+
+    b_line = spherely.linestring([(1.0, 1.0), (1.0, 2.0)])
+    assert spherely.touches(a_p, b_line)
+
+
+@pytest.fixture
+def parent_poly():
+    return spherely.polygon(
+        [
+            (-118.0, 60.0),
+            (-118.0, 40.0),
+            (-118.0, 23.0),
+            (34.0, 23.0),
+            (34.0, 40.0),
+            (34.0, 60.0),
+        ]
+    )
+
+
+@pytest.fixture
+def polys_to_check_for_covers():
+    return (
+        np.array(
+            [
+                # Basic point covers tests, outside, on boundary and interior
+                spherely.point(-120.0, 70.0),
+                spherely.point(-118.0, 41.0),
+                spherely.point(-116.0, 37.0),
+                # Basic polyline tests, crossing, on boundary and interior
+                spherely.linestring([(-120.0, 70.0), (-116.0, 37.0)]),
+                spherely.linestring([(-118.0, 41.0), (-118.0, 23.0)]),
+                spherely.linestring([(-117.0, 39.0), (-115.0, 37.0)]),
+                # Basic polygon test, crossing, shared boundary and interior
+                spherely.polygon(
+                    [(-120.0, 41.0), (-120.0, 35.0), (-115.0, 35.0), (-115.0, 41.0)]
+                ),
+                # TODO: This case is currently not fully correct. Supplying a polygon `a` and `b` and checking
+                # whether `a` covers `b`, where `a` and `b` share co-linear edges only works when the edges
+                # between them are identical. An example of this breaking would be a polygon `a` consisting of the edges AB, BC and CA and
```

Comment:
Can you reduce the line length here for the comments? (that makes it a bit more readable)

Reply:
Done